codegen/build-oas.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+
+set -eux -o pipefail
+
+version=$1 # e.g. 2024-07
+modules=("control")
+
+destination="src/main/java/org/openapitools"
+build_dir="gen"
+
+update_apis_repo() {
+	echo "Updating apis repo"
+	pushd codegen/apis
+		git fetch
+		git pull
+		just build
+	popd
+}
+
+verify_spec_version() {
+	local version=$1
+	echo "Verifying spec version $version exists in apis repo"
+	if [ -z "$version" ]; then
+		echo "Version is required"
+		exit 1
+	fi
+
+	verify_directory_exists "codegen/apis/_build/${version}"
+}
+
+verify_file_exists() {
+	local filename=$1
+	if [ ! -f "$filename" ]; then
+		echo "File does not exist at $filename"
+		exit 1
+	fi
+}
+
+verify_directory_exists() {
+	local directory=$1
+	if [ ! -d "$directory" ]; then
+		echo "Directory does not exist at $directory"
+		exit 1
+	fi
+}
+
+generate_client() {
+	local module_name=$1
+
+	oas_file="codegen/apis/_build/${version}/${module_name}_${version}.oas.yaml"
+	
+	verify_file_exists $oas_file
+
+	# Cleanup previous build files
+	echo "Cleaning up previous build files"
+	rm -rf "${build_dir}"
+
+	# Generate client module
+	docker run --rm -v $(pwd):/workspace openapitools/openapi-generator-cli:v7.0.1 generate \
```

Comment:
Is there a reason for using different openapi-generator version, compared to python's client generation script?
https://github.com/pinecone-io/pinecone-python-client/pull/361/files#diff-cae10a8f6beee215d2e44a3d55d2797a4e0844ea00bcb2b40bd53a76fb345da5R71

Reply:
The generator cli docker image bakes in language-specific code generation logic so it's not surprising that we might need different versions for different projects. The python sdk is stuck on an older version (v5) because in v6 of the generator cli a lot of breaking changes were made in the python generator that we aren't ready to consume in the sdk because it would create breaking changes for our users. For Java, we don't have that same constraint and since we did previous rounds of generation with a v7 generator CLI I put the same in this script to keep the code diff in these generated files to a minimum.